src/java/main/org/apache/zookeeper/server/ZooKeeperServerMain.java

```diff
@@ -146,14 +151,19 @@ public void runFromConfig(ServerConfig config) throws IOException, AdminServerEx
             );
             containerManager.start();
 
+            // Watch status of ZooKeeper server. It will do a graceful shutdown
+            // if the server is not running or hits an internal error.
+            healthMonitorLatch.await();
```

Comment:
Small point, but I'd rather call it "shutdownLatch" or "stopLatch" rather than "healthMonitorLatch". It is not only about the health of the server, the server might be completely healthy and you still want to cross the latch to stop it, right?